### PR TITLE
API Updates to maximal usable without code changes

### PR DESCRIPTION
**Pull request by bekuno**
API Updates to maximal usable without code changes:
(see also #7350)

- gradle-wrapper.properties:
  - update to the latest gradle plugin of 4.10.x

- main\build.gradle
  - org.apache.commons:
	- commons-collections4:4.2
    - commons-compress:1.18
    - commons-io:2.6
    - commons-lang3:3.8.1
   - org.assertj: Patch 2.9.1
  - com.fasterxml.jackson.core: Patch 2.9.6
  - com.google.android.gms:10.0.1 (latest)
  - io.reactivex.rxjava2: (latest)
	- rxjava:2.2.7
	- rxandroid:2.1.1
  - com.android.support.test:runner:1.0.2 (latest)

- build.gradle
  - Update compatible to AS 3.2.2:
    - com.android.tools.build:gradle:3.3.2'
  - other plugins as max as possible
    - com.github.ben-manes:gradle-versions-plugin:0.21.0'
    - de.mobilej.unmock:UnMockPlugin:0.7.3'
    - com.google.gms:google-services:2.1.2'
    - com.getkeepsafe.dexcount:dexcount-gradle-plugin:0.8.6'
    - com.gradle:build-scan-plugin:1.16'
  - remove some outdated checkes
  - Configuration (dependencyUpdates) for gradle-versions-plugin
	(gradleReleaseChannel starting 0.21.0)

### CODE REVIEW

**Comment by bekuno** [comment]
@kumy can you update the environment in the next day for the new packages?

**Comment by kumy** [comment]
@bekuno May you propose a PR on https://github.com/cgeo/cgeo-executor/blob/master/android-packages ?

Looking at the errors, they don't seems to be Android packages, but libraries, no?

**Comment by bekuno** [comment]
Yes, only libraries.
So normaly nothing to do for you?

It is strange that jenkins gives such errors:
  Could not resolve com.android.support:support-annotations:24.2.1
This is a standard condition.
Maybe it is a problem with different library versions. I have also this error:
  Dependency path 'cgeo:main:unspecified' --> 'com.android.support.test:runner:1.0.2' -->   'com.android.support.test:monitor:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
But there was no such error visible in AS. 
I will check that tomorrow.

**Comment by kumy** [comment]
IINM, gradle is responsible for downloading such libraries.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-all.zip
```
Comment (Bananeweizen) [inline gradle/wrapper/gradle-wrapper.properties]: It is not sufficient to just change the URL. You also need to update the wrapper script itself. See https://gradle.org/install/#with-the-gradle-wrapper
When you execute that, both the properties and the wrapper should be updated.

Comment (bekuno) [reply gradle/wrapper/gradle-wrapper.properties]: What script (gradlew ?) ant which point should be updated?

Comment (bekuno) [reply gradle/wrapper/gradle-wrapper.properties]: Ok, will do it.

**Pull request merged**